### PR TITLE
Restore reproducible libs

### PR DESCRIPTION
**Pull request by riverar**

### CODE REVIEW

File: crates/tools/msvc/src/main.rs [hunk 1]
```diff
@@ -67,7 +67,7 @@ fn main() {
             break;
         }
     }
-
+    archive.flush();
```
Comment (kennykerr) [inline crates/tools/msvc/src/main.rs:70]: May want to call `drop(archive)` right after this line to ensure the file is closed after flushing and before copying.

Comment (kennykerr) [reply crates/tools/msvc/src/main.rs:70]: I think the gnu tool does call `drop` but it fails to call `flush`.

**Review by kennykerr — DISMISSED** [review]

**Comment by riverar** [comment]
Investigating gnu lib reproducibility issue, see if we can get two birds with one PR here.

**Comment by riverar** [comment]
OK @kennykerr looking good. Manually verified the msvc/gnu libs against the workflow ones and they match now. We'll revisit and hopefully clean all this up soon.

**Review by kennykerr — DISMISSED** [review]

**Comment by kennykerr** [comment]
A MacOS concurrency limit is holding things up - since everything else has passed I'm going to cancel the run and complete the PR to relieve some pressure.

**Pull request merged**